### DISCUSSION TITLE
Failed to download Valheim server from Steam

### DISCUSSION THREAD

**Discussion by noonesaid** [lloesche/valheim-server-docker#223] [General]
I tried installing Valheim-server-docker via Docker on Windows today but the log shows an error:
"Failed to download Valheim server from Steam - retrying later - check your networking and volume access permissions"

Is anyone else getting this error?

**Comment by lloesche** [comment]
What is the `steamcmd` output saying? Paste the whole log, from start to failure.

**Reply by gabelucci** [reply]
Not the OP but I am having the same issue:

> INFO - Setting timezone Etc/UTC
INFO - Setting up syslogd - logging to stdout
2021-03-28 00:31:34,804 INFO Included extra file "/usr/local/etc/supervisor/conf.d/syslog.conf" during parsing
2021-03-28 00:31:34,804 INFO Set uid to user 0 succeeded
2021-03-28 00:31:34,809 INFO RPC interface 'supervisor' initialized
2021-03-28 00:31:34,809 INFO supervisord started with pid 1
2021-03-28 00:31:35,812 INFO spawned: 'syslogd' with pid 17
2021-03-28 00:31:35,813 INFO spawned: 'crond' with pid 18
2021-03-28 00:31:35,815 INFO spawned: 'valheim-bootstrap' with pid 19
Mar 28 00:31:35 syslogd started: BusyBox v1.32.1
2021-03-28 00:31:35,817 INFO success: valheim-bootstrap entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
Mar 28 00:31:35 crond[18]: crond (busybox 1.32.1) started, log level 8
Mar 28 00:31:35 /supervisord: valheim-bootstrap DEBUG - [19] - Running commit 182fd562331b542dc63133ce8e94038ac41dcb54
Mar 28 00:31:35 /supervisord: valheim-bootstrap DEBUG - [19] - Creating cron to do world backups using schedule 0 * * * *
Mar 28 00:31:35 /supervisord: valheim-bootstrap DEBUG - [19] - Creating cron to check for updates using schedule */15 * * * *
Mar 28 00:31:35 /supervisord: valheim-bootstrap DEBUG - [19] - Creating cron to restart valheim-server using schedule 0 5 * * *
2021-03-28 00:31:36,010 INFO spawned: 'valheim-updater' with pid 29
Mar 28 00:31:36 /supervisord: valheim-updater DEBUG - [29] - Running Valheim Server updater as user root uid 0
Mar 28 00:31:36 /supervisord: valheim-updater DEBUG - [29] - Found CPU with 1596.178 MHz
Mar 28 00:31:36 /supervisord: valheim-updater DEBUG - [29] - Memory total/free/available: 49439304/47045316/48416828
Mar 28 00:31:36 /supervisord: valheim-updater DEBUG - [29] - Valheim updater was just started - skipping connected players check
Mar 28 00:31:36 /supervisord: valheim-updater INFO - Downloading/updating/validating Valheim server from Steam
2021-03-28 00:31:37,251 INFO success: syslogd entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2021-03-28 00:31:37,251 INFO success: crond entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2021-03-28 00:31:37,251 INFO success: valheim-updater entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
Mar 28 00:31:37 /supervisord: valheim-bootstrap valheim-updater: started
2021-03-28 00:31:37,447 INFO spawned: 'valheim-backup' with pid 61
Mar 28 00:31:37 /supervisord: valheim-backup INFO - Backing up Valheim server worlds to /config/backups/worlds-20210328-003137.zip
Mar 28 00:31:37 /supervisord: valheim-backup   adding: worlds/ (stored 0%)
Mar 28 00:31:37 /supervisord: valheim-backup   adding: worlds/steam_autocloud.vdf
Mar 28 00:31:37 /supervisord: valheim-backup  (stored 0%)
Mar 28 00:31:37 /supervisord: valheim-backup   adding: worlds/Boofit.db
2021-03-28 00:31:38,465 INFO success: valheim-backup entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
Mar 28 00:31:38 /supervisord: valheim-bootstrap valheim-backup: started
2021-03-28 00:31:38,483 INFO exited: valheim-bootstrap (exit status 0; expected)
Mar 28 00:31:38 /supervisord: valheim-backup  (deflated 69%)
Mar 28 00:31:38 /supervisord: valheim-backup   adding: worlds/Boofit.fwl
Mar 28 00:31:38 /supervisord: valheim-backup  (deflated 5%)
Mar 28 00:31:38 /supervisord: valheim-backup   adding: worlds/Boofit.fwl.old (deflated 5%)
Mar 28 00:31:38 /supervisord: valheim-backup   adding: worlds/Boofit.db.old
Mar 28 00:31:39 /supervisord: valheim-backup  (deflated 69%)
Mar 28 00:31:39 /supervisord: valheim-backup INFO - Removing backups older than 3 days
Mar 28 00:31:42 /supervisord: valheim-updater Redirecting stderr to '/root/Steam/logs/stderr.txt'
Mar 28 00:31:42 /supervisord: valheim-updater [  0%] Checking for available updates...
Mar 28 00:31:42 /supervisord: valheim-updater [----] Verifying installation...
Mar 28 00:31:42 /supervisord: valheim-updater [  0%] Downloading update...
Mar 28 00:31:42 /supervisord: valheim-updater [  0%] Checking for available updates...
Mar 28 00:31:42 /supervisord: valheim-updater [----] Download complete.
Mar 28 00:31:42 /supervisord: valheim-updater [----] Extracting package...
Mar 28 00:31:42 /supervisord: valheim-updater [----] Extracting package...
Mar 28 00:31:42 /supervisord: valheim-updater [----] Extracting package...
Mar 28 00:31:42 /supervisord: valheim-updater [----] Extracting package...
Mar 28 00:31:42 /supervisord: valheim-updater [----] Installing update...
Mar 28 00:31:42 /supervisord: valheim-updater [----] Installing update...
Mar 28 00:31:42 /supervisord: valheim-updater [----] Installing update...
Mar 28 00:31:42 /supervisord: valheim-updater [----] Installing update...
Mar 28 00:31:42 /supervisord: valheim-updater [----] Installing update...
Mar 28 00:31:42 /supervisord: valheim-updater [----] Installing update...
Mar 28 00:31:42 /supervisord: valheim-updater [----] Installing update...
Mar 28 00:31:42 /supervisord: valheim-updater [----] Installing update...
Mar 28 00:31:42 /supervisord: valheim-updater [----] Cleaning up...
Mar 28 00:31:42 /supervisord: valheim-updater [----] Update complete, launching Steamcmd...
Mar 28 00:31:43 /supervisord: valheim-updater Redirecting stderr to '/root/Steam/logs/stderr.txt'
Mar 28 00:31:43 /supervisord: valheim-updater [  0%] Checking for available updates...
Mar 28 00:31:43 /supervisord: valheim-updater [----] Verifying installation...
Mar 28 00:31:43 /supervisord: valheim-updater Steam Console Client (c) Valve Corporation
Mar 28 00:31:43 /supervisord: valheim-updater -- type 'quit' to exit --
Mar 28 00:31:43 /supervisord: valheim-updater Loading Steam API...OK.
Mar 28 00:31:43 /supervisord: valheim-updater
Mar 28 00:31:43 /supervisord: valheim-updater Connecting anonymously to Steam Public...
Mar 28 00:31:43 /supervisord: valheim-updater Logged in OK
Mar 28 00:31:43 /supervisord: valheim-updater Waiting for user info...
Mar 28 00:31:46 /supervisord: valheim-updater OK
Mar 28 00:31:46 /supervisord: valheim-updater  Update state (0x3) reconfiguring, progress: 0.00 (0 / 0)
Mar 28 00:31:46 /supervisord: valheim-updater Error! App '896660' state is 0x202 after update job.
Mar 28 00:31:46 /supervisord: valheim-updater ERROR - Failed to download Valheim server from Steam - retrying later - check your networking and volume access permissions

**Reply by lloesche** [reply]
So that `Error! App '896660' state is 0x202 after update job.` is what Steam outputs. I did a quick Google search and it all points to a full disk error. Or Steam being unable to figure out how much disk space is free. We had a similar report of a QNAP server using ZFS. Apparently Steam and ZFS don't play nice in some configurations. Although I tried it on my server and had no issues.
I'm currently pushing a commit that will output the storage configuration on each update check just like it already outputs the CPU and RAM configuration. That should make it a bit easier to spot storage issues.

**Reply by inyfinn** [reply]
Hello.

![image](https://user-images.githubusercontent.com/80358004/171844391-64d54278-f9bc-4d59-9671-294b1b7d355d.png)

I have this problem. The image is unable to download valheim server files. I am getting error 0x406. I don't know what causes this error. I have a synology server. Two 4 TB RAID 1 hard drives.

**Reply by an-englishman** [reply]
I also have this problem. Was working but after I updated my Synology NAS it cant download Valheim server.
